syntax="proto3";

package clone.data_types;

enum ErrorType{
  UNKNOWN = 0;
  GOLEM_ERROR = 1;
  WRONG_REQUEST = 2;
  DISABLED_FUNCTIONALITY = 3;
}

message ErrorInfo{
  ErrorType error=1;
  string info=2;
  optional uint32 subtype=3;
}

message ServerResponse{
  bool success=1;
  optional ErrorInfo error=2;
}
